### ISSUE TITLE
Add "debug_traceCall" JSON-RPC method

### ISSUE THREAD

**Issue by LuozhuZhang** [NomicFoundation/hardhat#2971]
Edit by @fvictorio, see original issue below.

We should either do this after [Basic support for custom javascript tracers in debug_traceTransaction](https://github.com/NomicFoundation/hardhat/issues/4234), or in parallel to that and then add support for js tracers to `debug_traceCall` as a separate effort.

To-do:
- [x] Implement `debug_traceCall`
- [x] Manually compare some executions of it with geth
- [x] Add tests
- [x] Add the method to the list of supported methods in the docs

---

Not Support now
```
"data":{"message":"Method debug_traceCall not found"}
```

Check on [geth](https://geth.ethereum.org/docs/rpc/ns-debug#debug_tracecall)

Search terms: traceCall trace call

**Comment by unparalleled-js** [comment]
This would be really helpful! :)

**Comment by ChristopherDedominici** [comment]
RPC method debug_traceCall has been implemented

**Comment by ChristopherDedominici** [comment]
RPC method debug_traceCall has been implemented

**Comment by fvictorio** [comment]
Released in [v2.17.4](https://github.com/NomicFoundation/hardhat/releases/tag/hardhat%402.17.4) :tada:
